SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialMetrics.cpp

```diff
@@ -60,3 +68,156 @@ double USpatialMetrics::CalculateLoad() const
+		{
+			if (A.Type != B.Type)
+			{
+				return static_cast<int>(A.Type) < static_cast<int>(B.Type);
+			}
+			return A.Calls > B.Calls;
+		});
+
+		int MaxRPCNameLen = 0;
+		for (RPCStat& Stat : RecentRPCArray)
+		{
+			MaxRPCNameLen = FMath::Max(MaxRPCNameLen, Stat.Name.Len());
+		}
+
+		int TotalCalls = 0;
+		int TotalPayload = 0;
+
+		UE_LOG(LogSpatialMetrics, Log, TEXT("---------------------------"));
+		UE_LOG(LogSpatialMetrics, Log, TEXT("Recently sent RPCs - %s:"), NetDriver->IsServer() ? TEXT("Server") : TEXT("Client"));
+		UE_LOG(LogSpatialMetrics, Log, TEXT("RPC Type           | %s | # of calls |  Calls/sec | Total payload | Avg. payload | Payload/sec"), *FString(TEXT("RPC Name")).RightPad(MaxRPCNameLen));
+
+		FString SeparatorLine = FString::Printf(TEXT("-------------------+-%s-+------------+------------+---------------+--------------+------------"), *FString::ChrN(MaxRPCNameLen, '-'));
+
+		ESchemaComponentType PrevType = SCHEMA_Invalid;
+		for (RPCStat& Stat : RecentRPCArray)
+		{
+			FString RPCTypeField;
+			if (Stat.Type != PrevType)
+			{
+				RPCTypeField = RPCSchemaTypeToString(Stat.Type);
+				PrevType = Stat.Type;
+				UE_LOG(LogSpatialMetrics, Log, TEXT("%s"), *SeparatorLine);
+			}
+			UE_LOG(LogSpatialMetrics, Log, TEXT("%s | %s | %10d | %10.4f | %13d | %12.4f | %11.4f"), *RPCTypeField.RightPad(18), *Stat.Name.RightPad(MaxRPCNameLen), Stat.Calls, Stat.Calls / TrackRPCInterval, Stat.TotalPayload, (float)Stat.TotalPayload / Stat.Calls, Stat.TotalPayload / TrackRPCInterval);
+			TotalCalls += Stat.Calls;
+			TotalPayload += Stat.TotalPayload;
+		}
+		UE_LOG(LogSpatialMetrics, Log, TEXT("%s"), *SeparatorLine);
+		UE_LOG(LogSpatialMetrics, Log, TEXT("Total              | %s | %10d | %10.4f | %13d | %12.4f | %11.4f"), *FString::ChrN(MaxRPCNameLen, ' '), TotalCalls, TotalCalls / TrackRPCInterval, TotalPayload, (float)TotalPayload / TotalCalls, TotalPayload / TrackRPCInterval);
+
+		RecentRPCs.Empty();
+	}
+
+	bRPCTrackingEnabled = false;
+
+	// If RPC tracking is stopped on a client, send a command to the server to stop tracking.
+	if (!NetDriver->IsServer())
+	{
+		FUnrealObjectRef PCObjectRef = NetDriver->PackageMap->GetUnrealObjectRefFromObject(Cast<APlayerController>(NetDriver->GetSpatialOSNetConnection()->OwningActor));
+		Worker_EntityId ControllerEntityId = PCObjectRef.Entity;
+
+		if (ControllerEntityId != SpatialConstants::INVALID_ENTITY_ID)
+		{
+			Worker_CommandRequest Request = {};
+			Request.component_id = SpatialConstants::DEBUG_METRICS_COMPONENT_ID;
+			Request.schema_type = Schema_CreateCommandRequest(SpatialConstants::DEBUG_METRICS_COMPONENT_ID, SpatialConstants::DEBUG_METRICS_STOP_RPC_METRICS_ID);
+			NetDriver->Connection->SendCommandRequest(ControllerEntityId, &Request, SpatialConstants::DEBUG_METRICS_STOP_RPC_METRICS_ID);
+		}
+		else
+		{
+			UE_LOG(LogSpatialMetrics, Warning, TEXT("SpatialStopRPCMetrics: Could not resolve local PlayerController entity!"));
```

Comment:
Should probably also mention this means we haven't been able to turn off tracking on the server.